tests/yast2_gui/yast2_software_management.pm

```diff
@@ -9,16 +9,124 @@
+    wait_still_screen 60;
+
+    # Package - All packages - Update
+    assert_and_click 'sw_package';
+    # use mouse_set and click All in the List because assert_and_click won't hold
+    # the menu open for update if newer version available
+    mouse_set(184, 354);
+    mouse_click('left', 2);
+    assert_and_click 'update-if-newer-version-available';
+
+    # Configurations - Repositories, Online update
+    assert_and_click 'sw_configurations';
+    assert_and_click 'configurations_repo';
+    assert_screen 'configured_sw_repo', 90;
+    # check PGP keys
+    send_key 'alt-g';
+    assert_screen 'gpg-public-key-management';
+    # cancel here and go back to main menu
+    assert_and_click 'configured_sw_gpg_cancel';
+    assert_and_click 'configured_sw_repo_cancel';
+
+    # Dependencies
+    assert_and_click 'sw_dependencies';
+    assert_and_click 'check_now';
+    assert_and_click 'all-packages-dependencies-ok';
+
+    # Options - Cleanup when deleting packages
+    assert_and_click 'sw_options';
+    assert_and_click 'cleanup-when-deleting-packages';
+
+    # Extras - Show Products, Changes, History
+    assert_and_click 'sw_extras';
+    assert_and_click 'show-products';
+    assert_and_click 'Products-ok';
+    assert_and_click 'sw_extras';
+    assert_and_click 'show-changes';
+    assert_and_click 'changed-packages-ok';
+    assert_and_click 'sw_extras';
+    assert_and_click 'show-history';
+    assert_and_click 'History-close';
+
+    # Help - Overview, Symbol, keys
+    assert_and_click 'sw_help';
+    assert_and_click 'help_overview';
+    assert_and_click 'overview-ok';
+    assert_and_click 'sw_help';
+    assert_and_click 'help-symbols';
+    assert_and_click 'symbols-ok';
+    assert_and_click 'sw_help';
+    assert_and_click 'special-keys-overview';
+    assert_and_click 'special-keys-overview-ok';
+
+    # View - Pattern
+    assert_and_click 'sw_view';
+    assert_and_click 'view_patterns';
+    assert_screen 'pattern-list';
+
+    # select a package to install: apache2
+    assert_and_click 'sw_search';
+    mouse_set(144, 183);
+    mouse_click;
+    type_string 'apache2';
+    send_key 'alt-s';
+    assert_and_click 'select-and-install';
+    assert_screen 'package-selected';
+
+    # RPM Groups - check Package Groups
+    assert_and_click 'rpm-groups';
+    assert_screen 'groups-list';
+
+    # Installation Summary
+    assert_and_click 'installation-summary';
+    assert_screen 'summary_details';
+
+    # Cover cases like automatic changes and unsupported packages
+    send_key "alt-a";
+    wait_still_screen;
+    send_key 'alt-f';
+    assert_screen [qw(automatic-changes sw_management)];
+    if (match_has_tag('automatic-changes')) {
+        send_key 'alt-o';
+        assert_screen [qw(unsupported-packages sw_management)];
+        if (match_has_tag('unsupported-packages')) {
+            send_key 'alt-o';
+        }
+        assert_screen 'installation-report', 180;
```

Comment:
So we assert sw_management and if it matches we assert installation-report even if we didn't make any step. So should it be installation-report or they are same and then we need to assert again only if ```match_has_tag('unsupported-packages')``` is true.

Reply:
will change here as well: assert_screen  [qw(unsupport-packages installation-report)]

and take send_key 'alt-f' out of if {}